File: public/docs/_examples/intro/dart/lib/hero_service.dart
library developer_guide_intro.hero_service;

import 'package:angular2/angular2.dart';
import 'package:developer_guide_intro/hero.dart';
import 'package:developer_guide_intro/backend_service.dart';
import 'package:developer_guide_intro/logger_service.dart';

@Injectable()
class HeroService {
  var _backendService;
  var _logger;
  HeroService(){
    _logger = new Logger();
    _backendService = new BackendService();
  }
  getHeroes(){
    List<Hero> heroes = _backendService.getAll(Hero);
    return heroes;
  }
}

File: public/docs/_examples/router/ts/app/heroes/hero.service.ts
// #docregion
import {Injectable} from 'angular2/core';

export class Hero {
  constructor(public id: number, public name: string) { }
}

@Injectable()
export class HeroService {
  getHeroes() { return heroesPromise; }

  getHero(id: number | string) {
    return heroesPromise
      .then(heroes => heroes.filter(h => h.id === +id)[0]);
  }
}

var HEROES = [
	new Hero(11, 'Mr. Nice'),
	new Hero(12, 'Narco'),
	new Hero(13, 'Bombasto'),
	new Hero(14, 'Celeritas'),
	new Hero(15, 'Magneta'),
	new Hero(16, 'RubberMan')
];

var heroesPromise = Promise.resolve(HEROES);
